Subject: Fixed documentation on automatic conversion of Boolean
 Note: This bug has been forwarded upstream and rejected. As a reviewer
 it seems to me that this rejection must be due to a misunderstanding.
 As such the patch should stay but an upstream resolution should not be 
 expected any time soon.
Author: Jozef Kutej <jozef@kutej.net>
Reviewed-by: Nicholas Bamber <nicholas@periapt.co.uk>
Bug: http://rt.cpan.org/Public/Bug/Display.html?id=53975
Last-Update: 2010-09-06
--- a/XS.pm
+++ b/XS.pm
@@ -681,8 +681,9 @@
 returning the resulting simple scalar or reference. Croaks on error.
 
 JSON numbers and strings become simple Perl scalars. JSON arrays become
-Perl arrayrefs and JSON objects become Perl hashrefs. C<true> becomes
-C<1>, C<false> becomes C<0> and C<null> becomes C<undef>.
+Perl arrayrefs and JSON objects become Perl hashrefs.
+C<true> becomes JSON::XS::true (equals 1 numerically and as a string),
+C<false> becomes JSON::XS::false (equals 0) and C<null> becomes C<undef>.
 
 =item ($perl_scalar, $characters) = $json->decode_prefix ($json_text)
 
